SPIFlashLogger.class.nut

```diff
@@ -224,6 +224,68 @@ class SPIFlashLogger {
         });
     }
 
+    function readSync(index = 1) {
```

Comment:
The readSync code is essentially just a synchronous "unrolling" of the *read()* function.  Because both methods support reading from either the oldest or the newest log, index 0 is rather arbitrary and -0 is even more confusing...  By 1 indexing, the code can be consistent, rather you are scanning forward or reverse.